encrypter.cpp

```diff
@@ -0,0 +1,177 @@
+  out.write(tmp.c_str(), tmp.length());
+}
+
+//=====read_meta=====//
+std::streampos read_meta(std::ifstream& file)
+{
+  std::string tmp;
+  std::getline(file, tmp);
+  if (tmp.substr(0, 17) != "<encrypted> meta=")
+    put_error("A wrong file was given.\n");
+
+  return std::stol(tmp.substr(17, tmp.length())); 
+}
+
+//=====set_up_for_encrypting=====//
+void set_up_for_encrypting(
+    std::ofstream& out, const std::string& filename, std::streampos& key_starts, std::ifstream& file,
+    const std::string& keyname, std::ifstream& key)
+{
+  out.open(filename + ".enc", std::ios::out|std::ios::binary);
+
+  key_starts = read_key_position(keyname);
+  if ((long)(key.tellg()) - key_starts + 1 < file.tellg())
+    put_error("The key needs to be changed. There is not enough data to encrypt file safely.\n");
+
+  record_meta_to_both_files(key_starts, file.tellg(), keyname, out);
+  file.seekg(0, std::ios::beg);
+}
+
+//=====set_up_for_decrypting=====//
+void set_up_for_decrypting(
+    std::ofstream& out, const std::string& filename, std::streampos& key_starts, std::ifstream& file)
+{
+  file.seekg(0, std::ios::beg);
+  key_starts = read_meta(file);
+
+  out.open(filename.substr(0, filename.length() - 4), std::ios::out|std::ios::binary);
+}
+
+//=====crypting=====//
+void crypting(const std::string& filename, const std::string& keyname, bool encrypt = true)
+{
+  std::streampos key_starts;
+  std::ifstream file (filename, std::ios::in|std::ios::binary|std::ios::ate);
+  std::ifstream key (keyname, std::ios::in|std::ios::binary|std::ios::ate);
+
+  if (!file.is_open() || !key.is_open())
+  {
+    put_error("Couldn't open input and/or key files.\n");
+  }
+
+  std::ofstream out;
+
+  if (encrypt)
+    set_up_for_encrypting(out, filename, key_starts, file, keyname, key);
+  else
+    set_up_for_decrypting(out, filename, key_starts, file);
+
+  key.seekg(key_starts, std::ios::beg);
+  if (!out.is_open())
+    put_error("Couldn't open output file.\n");
+
+  Byte orig;
+  while (file.read(&orig, 1))
+  {
+    Byte code;
+    key.read(&code, 1);
+    Byte res = orig ^ code;
+
+    out.write(&res, 1);
+  }
+}
+
+//=====main=====//
+int main(int argc, Byte* argv[])
+{
+  try
+  {
+    if (argv[1][0] != '-')
+      put_error("A key should go first. For help use key '-h'.\n");
+
+    switch(argv[1][1])
+    {
+      case 'k':
+        make_key(argv[2]);
+        break;
+
+      case 'e':
+        crypting(argv[2], argv[3]);
+        break;
+
+      case 'd':
+        crypting(argv[2], argv[3], false);
+        break;
+
+      default:
+        std::cout << "To make a key use '-k' and provide a file as a base for your key.\nTo encrypt a file use key '-e', provide the file to be encrypted, and lastly a key file.\nTo decode a file use '-d', provide file to be decypted and a key that was used for enryption.\n";
+    }
+ }
+  catch(const std::exception& e)
+  {
+    std::cerr << "Exception: " << e.what() << std::endl;  
```

Comment:
If you use you put_error function it would be more consistent logic - every error goes through one function. And you can get rid of exit() call